### PR TITLE
Refactor widget module

### PR DESCRIPTION
**Pull request by samuelgarcia**
This a major refactoring of the spikeinterface.widgets modue.

* remove the intermediate Plotter for every backend (this make the debugging and wirtting more efficient)
* all backend are implemented with a `Widget.plot_matplotlib()`,  `Widget.plot_ipywidgets()`,  `Widget.plot_sortingview()`
* the import will be lazy (before the entire dependencies was importanted)

At the moment this is quite drafty but the proff of concept is OK I think.

I propose to sqtart with this and then we will be able to move `_legacy_mpl_widgets` into this new api.

This PR also change `plot_timeseries()` -> `plot_traces()`.

### CODE REVIEW

**Comment by samuelgarcia** [comment]
@alejoe91 I think this is ready to be merged.

File: src/spikeinterface/widgets/tests/test_widgets.py [hunk 1]
```diff
@@ -35,6 +36,7 @@
 else:
     cache_folder = Path("cache_folder") / "widgets"
 
+print(cache_folder)
```
Comment (alejoe91) [inline src/spikeinterface/widgets/tests/test_widgets.py:39]: remove

File: src/spikeinterface/widgets/tests/test_widgets.py [hunk 2]
```diff
@@ -13,7 +13,8 @@
 
 from spikeinterface import extract_waveforms, load_waveforms, download_dataset, compute_sparsity
 
-from spikeinterface.widgets import HAVE_MPL, HAVE_SV
+# from spikeinterface.widgets import HAVE_MPL, HAVE_SV
```
Comment (alejoe91) [inline src/spikeinterface/widgets/tests/test_widgets.py:16]: remove

File: src/spikeinterface/widgets/all_amplitudes_distributions.py [hunk 3]
```diff
@@ -47,3 +45,40 @@ def __init__(
         )
 
         BaseWidget.__init__(self, plot_data, backend=backend, **backend_kwargs)
+
+    def plot_matplotlib(self, data_plot, **backend_kwargs):
+        import matplotlib.pyplot as plt
+        from .utils_matplotlib import make_mpl_figure
```
Comment (alejoe91) [inline src/spikeinterface/widgets/all_amplitudes_distributions.py:51]: Can you move relative imports after the absolute ones?

File: src/spikeinterface/widgets/quality_metrics.py [hunk 4]
```diff
@@ -23,7 +22,7 @@ class QualityMetricsWidget(MetricsBaseWidget):
         For sortingview backend, if True the unit selector is not displayed, default False
     """
 
-    possible_backends = {}
+    # possible_backends = {}
```
Comment (alejoe91) [inline src/spikeinterface/widgets/quality_metrics.py:25]: remove

File: src/spikeinterface/widgets/sorting_summary.py [hunk 5]
```diff
@@ -34,7 +34,7 @@ class SortingSummaryWidget(BaseWidget):
         (sortingview backend)
     """
 
-    possible_backends = {}
+    # possible_backends = {}
```
Comment (alejoe91) [inline src/spikeinterface/widgets/sorting_summary.py:37]: remove

File: src/spikeinterface/widgets/spike_locations.py [hunk 6]
```diff
@@ -36,7 +35,7 @@ class SpikeLocationsWidget(BaseWidget):
         If True, the axis is set to off. Default False (matplotlib backend)
     """
 
-    possible_backends = {}
+    # possible_backends = {}
```
Comment (alejoe91) [inline src/spikeinterface/widgets/spike_locations.py:38]: remove

File: src/spikeinterface/widgets/template_similarity.py [hunk 7]
```diff
@@ -27,7 +25,7 @@ class TemplateSimilarityWidget(BaseWidget):
         If True, color bar is displayed, default True.
     """
 
-    possible_backends = {}
+    # possible_backends = {}
```
Comment (alejoe91) [inline src/spikeinterface/widgets/template_similarity.py:28]: remove

File: src/spikeinterface/widgets/unit_depths.py [hunk 8]
```diff
@@ -24,7 +24,7 @@ class UnitDepthsWidget(BaseWidget):
         Sign of peak for amplitudes, default 'neg'
     """
 
-    possible_backends = {}
+    # possible_backends = {}
```
Comment (alejoe91) [inline src/spikeinterface/widgets/unit_depths.py:27]: remove

File: src/spikeinterface/widgets/unit_summary.py [hunk 9]
```diff
@@ -31,7 +30,7 @@ class UnitSummaryWidget(BaseWidget):
         If WaveformExtractor is already sparse, the argument is ignored
     """
 
-    possible_backends = {}
+    # possible_backends = {}
```
Comment (alejoe91) [inline src/spikeinterface/widgets/unit_summary.py:33]: remove

File: src/spikeinterface/widgets/unit_locations.py [hunk 10]
```diff
@@ -31,7 +33,7 @@ class UnitLocationsWidget(BaseWidget):
         If True, the axis is set to off, default False (matplotlib backend)
     """
 
-    possible_backends = {}
+    # possible_backends = {}
```
Comment (alejoe91) [inline src/spikeinterface/widgets/unit_locations.py:36]: remove

**Review by alejoe91 — CHANGES_REQUESTED** [review]
@samuelgarcia looks good! Can you remove the commented stuff?

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**